polling_stations/apps/data_collection/management/commands/import_birmingham.py

```diff
@@ -1,43 +1,77 @@
 from data_collection.management.commands import BaseHalaroseCsvImporter
+from django.contrib.gis.geos import Point
 
 
 class Command(BaseHalaroseCsvImporter):
     council_id = "E08000025"
     addresses_name = (
-        "europarl.2019-05-23/Version 1/polling_station_export-2019-04-30.csv"
+        "parl.2019-12-12/Version 1/polling_station_export-2019-11-07brum.csv"
     )
     stations_name = (
-        "europarl.2019-05-23/Version 1/polling_station_export-2019-04-30.csv"
+        "parl.2019-12-12/Version 1/polling_station_export-2019-11-07brum.csv"
     )
-    elections = ["europarl.2019-05-23"]
-    csv_encoding = "windows-1252"
+    elections = ["parl.2019-12-12"]
+    allow_station_point_from_postcode = False
+
+    def station_record_to_dict(self, record):
+        rec = super().station_record_to_dict(record)
+
+        if record.pollingstationname == "St Mary and Ambrose Church Hall":
+            rec["postcode"] = "B5 7RA"
+            rec["location"] = Point(-1.904365, 52.458623, srid=4326)
+
+        if record.pollingstationnumber in ["145", "146"]:
+            rec[
+                "internal_council_id"
+            ] = f"{record.pollingstationnumber}-pentecostal-city-mission-church"
```

Comment:
This isn't doing what you want it to - that's why you've got >1500 addresses with invalid station refs when you run it.

Reply:
I was only looking at half the problem :man_facepalming: .